commit 1c3a71babc503bac7e92828a6f96d0538fc16ce5
Author: Andrew Roetker <andrew.roetker@puppetlabs.com>
Date:   Mon Apr 10 15:29:54 2017 -0700

    Update parse util to understand header from multiple responses

diff --git a/lib/faraday/utils.rb b/lib/faraday/utils.rb
index 1cd6526..9f17c9b 100644
--- a/lib/faraday/utils.rb
+++ b/lib/faraday/utils.rb
@@ -89,7 +89,13 @@ module Faraday
 
       def parse(header_string)
         return unless header_string && !header_string.empty?
-        header_string.split(/\r\n/).
+
+        headers = header_string.split(/\r\n/)
+
+        # Break headers into distinct responses, and only consider the last response.
+        last_response = headers.slice_before(/^HTTP\//).to_a.last
+
+        last_response.
           tap  { |a| a.shift if a.first.index('HTTP/') == 0 }. # drop the HTTP status line
           map  { |h| h.split(/:\s+/, 2) }.reject { |p| p[0].nil? }. # split key and value, ignore blank lines
           each { |key, value|
